File: gherkin/dotnet/Gherkin.Specs/Helper/LineEndingHelper.cs
﻿namespace Gherkin.Specs.Helper
{
    public static class LineEndingHelper
    {
        public static string NormalizeLineEndings(string text)
        {
            return text.Replace("\r\n", "\n").TrimEnd('\n');
        }

        public static string StripLineEndings(string text)
        {
            return text.Replace("\r", "").Replace("\n", "").Trim();
        }
    }
}

File: gherkin/dart/lib/src/helpers/LineEndingHelper.dart
import 'package:gherkin/extensions.dart';

class LineEndingHelper
{
  static String normalizeLineEndings(String text)
  {
    if(text.isNotEmpty) {
      text = text.replaceAll('\r\n', '\n').trimEnd('\n');
    }
    return text;
  }

  static String stripLineEndings(String text)
  {
    return text.replaceAll('\r', '').replaceAll('\n', '').trim();
  }
}
